packages/ckeditor5-table/src/tablecolumnresize/converters.js

```diff
@@ -53,6 +53,10 @@ export function upcastColgroupElement( editor ) {
 			} )
 			.join( ',' );
 
+		if ( columnWidthsAttribute.match( 'auto' ) ) {
+			columnWidthsAttribute = normalizeColumnWidthsAttribute( columnWidthsAttribute ).map( width => width + '%' ).join( ',' );
```

Comment:
> Because I can see how we _could_ (but we don't want to spend time on it now) go extra mile here and calculate width only for `auto` fields and retain widths of the other fields.

Actually it works that way now. I added a unit test confirming it (`should adjust the missing column widths proportionally`).

It stands in the opposition to the docs of `normalizeColumnWidthsAttribute(`) function, so I'll revamp it to reflect the actual state.

Reply:
As for [https://github.com/ckeditor/ckeditor5/pull/11914#discussion_r900092803,](https://github.com/ckeditor/ckeditor5/pull/11914#discussion_r900092803,) this was a bit more tangled as it required changing the parameter type in the `normalizeColumnWidthsAttribute()` function, adjusting other code using it, and in the end I decided to change its name as it no longer operates on the attribute (so a string) but on the array of widths. I found it unintuitive that this function takes a string and returns an array, so now it should be a bit more readable.